ToolBox/FactorFileGenerator.cs

```diff
@@ -226,11 +226,16 @@ private FactorFileRow CalculateNextDividendFactor(BaseData dividend, FactorFileR
 
             TradeBar previousClosingPrice = FindPreviousTradableDayClosingPrice(eventDayData.Time);
 
-            var priceFactor = previousFactorFileRow.PriceFactor - (dividend.Value / ((previousClosingPrice.Close) * previousFactorFileRow.SplitFactor));
+            // round split factor back to natural number
+            var splitFactorMultiplier = Math.Round(1 / previousFactorFileRow.SplitFactor);
+
+            // adjust the dividend for both price and split factors (!)
+            var priceFactor = previousFactorFileRow.PriceFactor - splitFactorMultiplier * dividend.Value *
+                              previousFactorFileRow.PriceFactor / previousClosingPrice.Close;
 
             return new FactorFileRow(
                 previousClosingPrice.Time,
-                priceFactor.RoundToSignificantDigits(7),
+                Math.Round(priceFactor, 7),
```

Comment:
Curious as to the reasoning behind this change?

Reply:
I was under impression that reference files were generated using Math.Round + FCL method is less error prone. I do realize now - thanks for detecting it - this was ```FactorFileRow.ToCsv()``` that did the roundings. Using RoundToSignificantDigits does make sense here indeed; in particular in roundings of SplitFactors that can have one or few zeros between the decimal point and first significant number ```0.0357143```

Reply:
That was my thinking as well, so we don't lose precision as we go back further in time on securities that have experiences many large splits